chat-plugins/uno.js

```diff
@@ -0,0 +1,694 @@
+			if (room.unoDisabled) return this.errorReply("UNO is currently disabled in this room.");
+			let cap = target ? parseInt(target, 10) : null;
+			if (room.game) return this.errorReply("There is already a game in progress in this room.");
+			if (cap && cap < 2) return this.errorReply("The player cap has to be at least 2.");
+
+			room.game = new Game(room, cap);
+			//start message;
+			room.game.init();
+		},
+		join: function (target, room, user) {
+			if (room.game && room.game.gameid === "uno") room.game.join(user);
+		},
+		start: function (target, room, user) {
+			if (!this.can("announce", null, room)) return false;
+			if (room.game && room.game.gameid === "uno") room.game.start(user);
+		},
+		end: function (target, room, user) {
+			if (!this.can("announce", null, room)) return false;
+			if (room.game && room.game.gameid === "uno") room.game.end();
+		},
+		leave: function (target, room, user) {
+			if (room.game && room.game.gameid === "uno" && !room.game.started) {
+				let success = room.game.removePlayer(user.userid);
+				if (!success) return user.sendTo(this.room, "You have not signed up for the game.");
+				let ip = user.latestIp;
+				delete room.game.joinedIps[ip];
+				room.add("|uhtml|init" + room.unoGameId + "|" + user.name + " has left the game.");
+			}
+		},
+		play: function (target, room, user) {
+			if (!room.game || room.game.gameid !== "uno") return false;
+			if (!target) return this.parse("/help uno play");
+			let params = target.split(" ");
+			let card = params.shift();
+			let change = null;
+			if (card.charAt(0) === "Z" && ["R", "Y", "B", "G"].indexOf(params[0]) > -1) {
+				change = params[0];
+			} else if (card.charAt(0) === "Z") {
+				let uhtmlid = room.unoGameId + "display" + room.game.id;
+				return user.sendTo(room, getColourChange("/uno play " + card, room.game.data[user.userid], uhtmlid));
+			}
+			room.game.play(user, card, change);
+		},
+		draw: function (target, room, user) {
+			if (room.game && room.game.gameid === "uno") room.game.draw(user);
+		},
+		pass: function (target, room, user) {
+			if (room.game && room.game.gameid === "uno") room.game.pass(user);
+		},
+		dq: function (target, room, user) {
+			if (!this.can("announce", null, room)) return false;
+			if (!target) return this.parse("/help uno dq");
+			if (room.game && room.game.gameid === "uno") room.game.disqualify(toId(target));
+		},
+		dqhelp: ["/uno dq [user] - disqualifies a user from the game."],
+		display: function (target, room, user) {
+			if (!room.game || room.game.gameid !== "uno" || user.userid !== room.game.player) return false;
+			room.game.clearGameScreen(user);
+			room.game.id++;
+			room.game.sendGameScreen(user);
+		},
+		uno: function (target, room, user) {
+			if (room.game && room.game.gameid === "uno" && room.game.started) {
+				room.game.parseUNO(user);
+			}
+		},
+		enable: function (target, room, user) {
+			if (!this.can('tournamentsmanagement', null, room)) return;
```

Comment:
I'm just copying the hangman permissions for this, since i think that means if the user is RO in that room....